src/main/kotlin/com/hikit/cai2hikit/TrailFetchJob.kt

```diff
@@ -1,30 +1,67 @@
 package com.hikit.cai2hikit
 
+import org.springframework.data.mongodb.repository.MongoRepository
+
+import com.hikit.cai2hikit.dto.IdToUpdateDate
 import com.hikit.cai2hikit.dto.Trail
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.http.MediaType
 import org.springframework.scheduling.annotation.Scheduled
 import org.springframework.stereotype.Service
 import org.springframework.web.client.RestClient
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+private const val dateTimeFormat = "yyyy-MM-dd HH:mm:ss"
 
 @Service
-class TrailFetchJob {
+class TrailFetchJob(
+        @Value("\${job.fetch.bblatmin}") val bblatmin: String,
+        @Value("\${job.fetch.bblatmax}") val bblatmax: String,
+        @Value("\${job.fetch.bblongmin}") val bblongmin: String,
+        @Value("\${job.fetch.bblongmax}") val bblongmax: String
+    ) {
+
+    @Autowired
+    private lateinit var trailRepository: TrailRepository
```

Comment:
E' necessario, ora che buttiamo giu' codice funzionante, iniziare a scrivere degli unit test. Scrivi almeno un paio di unit test che vanno a verificare la comunicazione tra classi.
1. Sposta RestClient all'interno di un'altra classe e fatti restituire un bean
2. Iniettalo dentro il Job
3. Usalo.

Nel test, 
1. Verifica che siano avvenute le chiamate che ci aspettiamo.
2. Usa i mock per simulare una risposta incongruente con il nostro mapping (es: data in un formato diverso o simili) e verifica che il programma non si rompi malamente - nel caso questo avvenga, dovremmo loggare l'errore e procedere con il mapping successivo.